### PR TITLE
Test only changed exercises on CI

### PR DESCRIPTION
**Pull request by lpil**
Building and testing all the exercises on CI takes a long time
and can cause CI to time out.
Rather than testing them all we can just test the ones that have
changed on the current branch against the master branch.

I've also included a tiny change to `Bob` to show the script working on
this branch.

### CODE REVIEW

File: bin/test-changed.sh [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+XPURESCRIPT=$(readlink -f "$XPURESCRIPT/..")
+cd "$XPURESCRIPT"
+
+declare -i TEST_RESULT=0
+FAILED_EXERCISES=''
+
+cd exercises
+
+FILES_CHANGED=$(
+  git diff --name-status master..HEAD |
```
Comment (paf31) [inline bin/test-changed.sh]: So in Travis, is this going to always compare against `master`?

Comment (lpil) [inline bin/test-changed.sh]: Yup. I couldn't work out if there was a way to get the SHA of the previous passing commit on master. Possibly could use their API.

Comment (paf31) [reply bin/test-changed.sh]: I was thinking we'd want to compare against whatever revision the commit was based against.

Comment (paf31) [reply bin/test-changed.sh]: In which case you might be able to use something like `HEAD~1`.

Comment (lpil) [reply bin/test-changed.sh]: A PR is always off of master is may not be one commit, so that would be less correct than comparing with master, no?

Comment (paf31) [reply bin/test-changed.sh]: So you want to know which branch the PR was opened against?

Comment (paf31) [reply bin/test-changed.sh]: You could try the `TRAVIS_BRANCH` environment variable.

**Comment by lpil** [comment]
Perhaps we don't need this any more? Other PRs seem to be passing. I'm unsure why.

I'll close this, we can reopen if this becomes a problem again.

**Pull request closed without merging**